src/plots/plots.js

```diff
@@ -2027,3 +2052,67 @@ plots.doCalcdata = function(gd, traces) {
         calcdata[i] = cd;
     }
 };
+
+plots.generalUpdatePerTraceModule = function(subplot, subplotCalcData, subplotLayout) {
+    var traceHashOld = subplot.traceHash,
```

Comment:
I always feel like the mutate-existing-data approach contributes a bit to fragility, but I understand why these things are as they are.